### PR TITLE
15568 move-away-from-gorp-to-sqlx-in-pull_request_store

### PR DESCRIPTION
**Pull request by kashifsoofi**
#### Summary
mattermod: move away from gorp to sqlx in pull_request_store.go

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/15568

#### JIRA
https://mattermost.atlassian.net/browse/MM-28977

I think there should be a separte issue to remove gorp from project, please let me know if you want to do it as pat of this PR.

### CODE REVIEW

**Review by agnivade — APPROVED** [review]
Nice work!

**Comment by agnivade** [comment]
> I think there should be a separte issue to remove gorp from project, please let me know if you want to do it as pat of this PR.

Sounds good. Would you be up for finishing that too? ;)

**Comment by kashifsoofi** [comment]
> 
> 
> > I think there should be a separte issue to remove gorp from project, please let me know if you want to do it as pat of this PR.
> 
> Sounds good. Would you be up for finishing that too? ;)

I can try

**Comment by agnivade** [comment]
Thanks @kashifsoofi ! Looks like a simple cleanup to me. All the remaining stuff is just in sql_store.go, and we just need to remove it.

**Comment by kashifsoofi** [comment]
> Thanks @kashifsoofi ! Looks like a simple cleanup to me. All the remaining stuff is just in sql_store.go, and we just need to remove it.

Should I push it here or are you creating a new issue?

**Comment by agnivade** [comment]
It does not need a new issue. Feel free to push it directly, or in a separate PR. Totally up to you.

**Comment by kashifsoofi** [comment]
👍 cool was not sure from you comment earlier, will push changes shortly

File: store/sql_store.go [hunk 1]
```diff
@@ -113,7 +88,7 @@ func (ss *SQLStore) Spinmint() SpinmintStore {
 }
 
 func (ss *SQLStore) DropAllTables() {
-	err := ss.master.TruncateTables()
+	err := multiExec(ss.db, []string{"TRUNCATE TABLE Issues", "TRUNCATE TABLE PullRequests", "TRUNCATE TABLE Spinmint"})
```
Comment (agnivade) [inline store/sql_store.go:91]: I don't see much benefit in refactoring this. It can be inside this method itself.

Comment (kashifsoofi) [reply store/sql_store.go]: Updated

File: store/sql_store.go [hunk 2]
```diff
@@ -113,9 +88,12 @@ func (ss *SQLStore) Spinmint() SpinmintStore {
...
 
 func (ss *SQLStore) DropAllTables() {
-	err := ss.master.TruncateTables()
-	if err != nil {
-		mlog.Error("failed to drop all tables", mlog.Err(err))
+	stmts := []string{"TRUNCATE TABLE Issues", "TRUNCATE TABLE PullRequests", "TRUNCATE TABLE Spinmint"}
+	for _, s := range stmts {
+		_, err := ss.dbx.Exec(s)
+		if err != nil {
+			mlog.Error("failed to drop all tables", mlog.Err(err))
```
Comment (agnivade) [inline store/sql_store.go:95]: `"failed to drop table", mlog.Err(err), mlog.String("table", s)`

**Review by agnivade — APPROVED** [review]
Nice work!

**Review by isacikgoz — APPROVED** [review]
Wow, nicely done @kashifsoofi! 🚀

**Pull request merged**